File: src/java/com/springbok/pattern/PatternSNOR605V01Test.java
package com.springbok.pattern;

import com.springbok.utility.TestUtility;
import junit.framework.Assert;

import java.util.HashMap;
import java.util.Map;

public class PatternSNOR605V01Test {
	private static final double GainMax_input = 41;
	private static final double Phi0_input = 2;

	private static final double[] Phi_input = new double[] { 0.2, 5, 10, 150 };

	private static final double[] G_expected = new double[] { 35.000000000000000, 11.000000000000000, 6.025749891599531,
			0 };
	private static final double[] Gx_expected = new double[] { 35.000000000000000, 11.000000000000000,
			6.025749891599531, 0 };

	public void test_gain() {
		PatternSNOR605V01 p = new PatternSNOR605V01(GainMax_input, Phi0_input);

		Gain[] gains = new Gain[Phi_input.length];
		Map<String, Object> options = new HashMap<String, Object>();
		options.put("GainMax", GainMax_input);
		options.put("DoValidate", false);
		for (int i = 0; i < gains.length; i++) {
			Gain g = p.gain(Phi_input[i], options);
			gains[i] = g;
		}

		for (int i = 0; i < gains.length; i++) {
			Assert.assertTrue(Math.abs(gains[i].G - this.G_expected[i]) < TestUtility.HIGH_PRECISION);
			Assert.assertTrue(Math.abs(gains[i].Gx - this.Gx_expected[i]) < TestUtility.HIGH_PRECISION);
		}
	}
}

File: src/matlab/com/springbok/pattern/PatternSNOR605V01Test.m
classdef PatternSNOR605V01Test < TestUtility
% Tests methods of PatternSNOR605V01 class.
  
% Copyright (C) 2016 Springbok LLC <http://springbok.io/> All rights reserved.

  properties (Constant = true)
    
    GainMax_input = 41;
    Phi0_input = 2;
    
    self = struct( ...
        'GainMax', 41, ...
        'Phi0', 2);
    
    Phi_input = [0.2; 5; 10; 150];
    
    G_expected = [ ...
        35.000000000000000
        11.000000000000000
        6.025749891599531
        0
                 ];
    Gx_expected = [ ...
        35.000000000000000
        11.000000000000000
        6.025749891599531
        0
                  ];
    
  end % properties (Constant = true)
  
  methods
    
    function this = PatternSNOR605V01Test(logFId, testMode)
    % Constructs a PatternSNOR605V01Test
    %
    % Parameters:
    %   logFId - Log file identifier
    %   testMode - Test mode, if 'interactive' then beeps and pauses
      
    % Invoke superclass constructor
      if nargin == 0
        superArgs = {};
        
      else
        superArgs{1} = logFId;
        superArgs{2} = testMode;
        
      end % if
      this@TestUtility(superArgs{:});
      
    end % PatternSNOR605V01Test()
    
    function test_PatternSNOR605V01(this)
    % Tests PatternSNOR605V01 method.
      
      p = PatternSNOR605V01(this.GainMax_input, this.Phi0_input);
      
      this.assert_true( ...
          'PatternSNOR605V01', ...
          'PatternSNOR605V01', ...
          this.HIGH_PRECISION_DESC, ...
          PatternUtility.compare_pattern(this.self, p));
      
    end % test_PatternSNOR605V01()
    
    function test_gain(this)
    % Tests gain method.
      
      warning('off');
      
      p = PatternSNOR605V01(this.GainMax_input, this.Phi0_input);
      
      [G_actual, Gx_actual] = p.gain(this.Phi_input);
      
      t(1) = max(abs(G_actual - this.G_expected) < this.HIGH_PRECISION);
      t(2) = max(abs(Gx_actual - this.Gx_expected) < this.HIGH_PRECISION);
      
      this.assert_true( ...
          'PatternSNOR605V01', ...
          'gain', ...
          this.HIGH_PRECISION_DESC, ...
          min(t));
      
      [G, Gx, fH] = p.gain([0:.001:180], 'PlotFlag', true);
      pause(1); close(fH);
      
      warning('on');
      
    end % test_gain()
    
  end % methods
  
end % classdef
